benchmark/distributed/worker.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+const yargs = require('yargs');
+
+const jstp = require('../..');
+const parseAddress = require('./parse-address');
+const statistics = require('../statistics');
+
+const args = yargs
+  .option('address', {
+    alias: 'a',
+    describe: 'Address of master server',
+    type: 'string',
+    default: 'tcp://localhost:3000',
+  })
+  .option('connection', {
+    alias: 'c',
+    describe: 'Amount of connections',
+    type: 'number',
+    default: 128,
+  })
+  .option('interval', {
+    alias: 'i',
+    describe: 'Interval of requests sent by connection in ms',
+    type: 'number',
+    default: 10,
+  })
+  .option('size', {
+    alias: 's',
+    describe: 'Size of argument to call remote method with',
+    type: 'number',
+    default: 0,
+  })
+  .option('report', {
+    alias: 'r',
+    describe: 'Report interval in ms',
+    type: 'number',
+    default: 10000,
+  })
+  .help()
+  .alias('help', 'h')
+  .argv;
+
+const {
+  address,
+  size: argumentSize,
+  connection: connectionAmount,
+  interval: requestInterval,
+  report: reportInterval,
+} = args;
+
+const [error, parsed] = parseAddress(address);
+if (error) {
+  console.error(`Invalid master address: ${error.message}`);
+  process.exit(1);
+}
+
+const { transport, connect: masterAddress } = parsed;
+
+// Prevent Node.js from throwing DEPTH_ZERO_SELF_SIGNED_CERT.
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+let api = null;
+const connections = new Array(connectionAmount);
+
+let requestTimer = null;
+let reportTimer = null;
+
+let reportCycleStartHR = null;
+
+let responseTimesHR = [];
+
+const argument = '0'.repeat(argumentSize);
+
+let requests = 0;
+
+const createConnection = (index, address) => {
+  jstp[transport].connectAndInspect(
+    'server', null, ['server'], ...address, (error, connection) => {
+      if (error) {
+        console.error(`Could not connect to server:\n${error}`);
+        return;
+      }
+
+      connections[index] = connection;
+    }
+  );
+};
+
+const report = () => {
+  const req = requests;
+  requests = 0;
+  const responseTimes = responseTimesHR.map(
```

Comment:
Any use of calculating time here instead of immediately after receiving the callback?

Reply:
Yes. This way we can use library that will provide precise integer operations, when we find one. If you calculate time immediately after receiving callback precision will be lost.